.github/workflows/build-pr.yml

```diff
@@ -88,3 +89,45 @@ jobs:
           name: e2e-logs
           path: |
             logs-*.json        
+  
+  build_test_aks_e2e:
+    name: "Build and run e2e Test on AKS"
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+        KUBERNETES_VERSION: ["1.24.6"]
+        GATEKEEPER_VERSION: ["3.10.0", "3.11.0"]
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Az CLI login
+        uses: azure/login@v1
+        with:
+          creds: '{"clientId":"${{ secrets.AZURE_CLIENT_ID }}","clientSecret":"${{ secrets.AZURE_CLIENT_SECRET }}","subscriptionId":"${{ secrets.AZURE_SUBSCRIPTION_ID }}","tenantId":"${{ secrets.AZURE_TENANT_ID }}"}'
```

Comment:
why is `AZURE_CLIENT_SECRET` required?

Reply:
if `AZURE_CLIENT_SECRET` is removed, I think the rest of values can just be variables in the workflow instead of secrets. For github secrets, they need to be made accessible to forks which is not recommended, so you can declare them as env vars (xref: https://github.com/Azure/azure-workload-identity/blob/main/.github/workflows/azwi-e2e.yaml#L18-L21)

Reply:
I tried OIDC login but failed to login in the beginning. Then I tried to client_secret which didn't work either. The root cause is the missing access to forked PR. I've updated it back to OIDC login now. Once the PR gets merged, we can see if it will work as expected, the federated credentials are already set up. 
<img width="821" alt="image" src="https://user-images.githubusercontent.com/6919333/213358895-005f241d-62f7-4726-99e2-c485f9d30169.png">